app.R

```diff
@@ -143,33 +177,23 @@ server <- function(session, input, output) {
     selected.coverage <- input$slider.coverage
     selected.study <-
       selected.study[`Coverage (collapsed)` >= selected.coverage[1] &
-                       `Coverage (collapsed)` <= selected.coverage[2],]
+                       `Coverage (collapsed)` <= selected.coverage[2], ]
     
     # Keep only the metrics that will be plotted
-    selected.study <- split(selected.study, by = "Type")
-    selected.type <- input$check.type
-    type.to.keep <- names(selected.study) %in% selected.type
-    return(selected.study[type.to.keep])
+    # As the data table contains info fields not part of input$check.type, take away metrics that will not be plotted
+    return(selected.study[,!setdiff(CONFIG.ALLPLOTS, input$check.type), with = FALSE])
   })
   
   # Create the plot
   output$plot <- renderPlotly({
-    req(input$check.type)
+    req(input$check.type, nrow(dt.to.plot()) > 0)
     
-    temp.to.plot <- dt.to.plot()
-    req(temp.to.plot)
-    
-    # Removes the cryptic error message if the parameters are such that there are no data points to print
-    # This is not caught by the req call because the temp.to.plot has elements, but all the elements are empty
-    anything.to.print <- all(sapply(temp.to.plot, nrow))
-    if (!anything.to.print) {
-      return(NULL)
-    }
```

Comment:
Did you mean to remove this block?

Reply:
Yes. This [new sanity check](https://github.com/oicr-gsi/acquacotta-shiny-run-report/pull/6/files/cccd867ea50b0f4bec584c04bb0530e29abebba1#diff-1e678274602e250345bb25d52e7c2d76R189) does this job now. It is a lot simpler now, as the temp.to.plot data table is no longer being split.